docgen/src/guides/custom-backend.md

```diff
@@ -0,0 +1,170 @@
+
+## On the backend: create the necessary routes
+
+This guide assumes that you’ve got an existing server running on [http://localhost:3000](http://localhost:3000) with the route `POST /search` that takes the default Algolia query parameters as JSON. This backend could be using the [JavasScript API client](https://www.algolia.com/doc/api-client/javascript/getting-started/) to query Algolia, on top of any other operations you want to perform.
+
+The `algoliasearch` package will allow you to query Algolia from your backend. Here's an example using [Express](https://expressjs.com/):
+
+```javascript
+// Instantiate an Algolia client
+const algoliasearch = require('algoliasearch');
+const algoliaClient = algoliasearch('YOUR_APP_ID', 'YOUR_SEARCH_API_KEY');
+
+// Add the search endpoint
+app.post('/search', async (req, res) => {
+  const requests = req.body;
+  const results = await algoliaClient.search(requests);
+  res.status(200).send(results);
+});
+```
+
+### Supporting *Search For Facet Values*
+
+Algolia offers the concept of [*Search For Facet Values*](https://www.algolia.com/doc/api-reference/api-methods/search-for-facet-values/?language=javascript). This enables your refinement lists to be searchable. If your frontend makes use of this feature, via a search box on top of refinement lists, you can create an endpoint `POST /sffv`. Add a new route to support that:
+
+```javascript
+app.post('/sffv', async (req, res) => {
+  const requests = req.body;
+  const results = await algoliaClient.searchForFacetValues(requests);
+  res.status(200).send(results);
+});
+```
+
+Once your proxy server is ready, we get back to the frontend and are going to create a search client able to communicate with this server.
+
+## On the frontend: call your new backend routes
+
+### Searching for results
+
+A search client is an object which implements the method `search()`, called every time the user searches and refines results.
+
+```javascript
+search: (requests: SearchRequest[]) => Promise<{ results: SearchResponse[] }>
+```
+
+Since our server accepts the InstantSearch format as an input, we will only need to pass these requests to our backend in this method, and return the response.
+
+```javascript
+const customSearchClient = {
+  search(requests) {
+    return fetch('http://localhost:3000/search', {
+      method: 'post',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(requests),
+    }).then(res => res.json());
+  }
+};
+```
+
+We use the [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) to query the server; make sure to take note of the [browser compatibility](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API#Browser_compatibility) before using it in production.
+
+> If you want to transform the data to be passed to your server, you can learn more about all the parameters that the `search()` method supports in the Search Client Specification.
```

Comment:
if this spec doesn't exist yet, What about "[Algolia documentation](https://www.algolia.com/doc/api-reference/api-parameters/?language=javascript#parameters-list)"